ch_tools/chadmin/internal/object_storage/collect_metadata.py

```diff
@@ -15,24 +15,29 @@ def collect_metadata(paths: Iterable[Path]) -> ObjectKeyToMetadata:
     """
     Return dictionary of parsed metadata from local disk with access by S3 object key.
     """
-    logging.debug("Metadata collecting...")
+    logging.debug("Metadata collecting for paths: %s", str(paths))
     res: ObjectKeyToMetadata = defaultdict(dict)
 
-    for path in paths:
-        if not path.exists():
-            logging.debug("Metadata path `%s` is not exist", path)
-
-    for file in chain.from_iterable(path.rglob("*") for path in paths if path.exists()):
-        if not file.is_file():
-            continue
-
-        try:
-            metadata = S3ObjectLocalMetaData.from_file(file)
-        except Exception:
+    for top_dir in paths:
+        if not top_dir.exists():
+            logging.debug("Metadata top_dir `%s` is not exist", top_dir)
             continue
 
-        for obj in metadata.objects:
-            res[obj.key][file] = metadata
+        for dirpath, _dirnames, filenames in os.walk(str(top_dir)):
```

Comment:
Use `os.walk()` because it swallows `OSError`s by default (including `FileNotFoundError`) and go ahead.